arch/riscv/core/pmp.c

```diff
@@ -576,7 +632,7 @@ void z_riscv_pmp_stackguard_prepare(struct k_thread *thread)
  *
  * This is called on every context switch.
  */
-void z_riscv_pmp_stackguard_enable(struct k_thread *thread)
+void z_riscv_pmp_kernelmode_enable(struct k_thread *thread)
 {
 	LOG_DBG("pmp_stackguard_enable for thread %p", thread);
```

Comment:
Could you also update the comments and log messages related to pmp_stackguard?

After renaming `z_riscv_pmp_stackguard_*` to `z_riscv_pmp_kernelmode_*`, these entries represent kernel mode PMP configuration, not just the stack guard.